auth-web/src/components/auth/account-settings/payment/AccountPaymentMethods.vue

```diff
@@ -103,269 +104,281 @@
+        state.errorText = 'Bank information validation failed'
+        if (verifyPad?.message?.length) {
+          let msgList = ''
+          verifyPad.message.forEach((msg) => {
+            msgList += `<li>${msg}</li>`
+          })
+          state.errorText = `<ul style="list-style-type: none;">${msgList}</ul>`
+        }
+        errorDialog.value.open()
+        return false
       }
-      this.$refs.errorDialog.open()
-      return false
     }
-  }
 
-  private cancel () {
-    this.initialize()
-  }
+    function cancel () {
+      initialize()
+    }
 
-  private async save () {
-    this.isBtnSaved = false
-    this.isLoading = true
-    let isValid = false
+    async function save () {
+      state.isBtnSaved = false
+      state.isLoading = true
+      let isValid = false
 
-    let createRequestBody: CreateRequestBody
+      let createRequestBody: CreateRequestBody
 
-    if (this.selectedPaymentMethod === PaymentTypes.PAD) {
-      isValid = await this.verifyPAD()
-      createRequestBody = {
-        paymentInfo: {
-          paymentMethod: PaymentTypes.PAD,
-          bankTransitNumber: this.padInfo.bankTransitNumber,
-          bankInstitutionNumber: this.padInfo.bankInstitutionNumber,
-          bankAccountNumber: this.padInfo.bankAccountNumber
+      if (state.selectedPaymentMethod === PaymentTypes.PAD) {
+        isValid = await verifyPAD()
+        createRequestBody = {
+          paymentInfo: {
+            paymentMethod: PaymentTypes.PAD,
+            bankTransitNumber: state.padInfo.bankTransitNumber,
+            bankInstitutionNumber: state.padInfo.bankInstitutionNumber,
+            bankAccountNumber: state.padInfo.bankAccountNumber
+          }
+        }
+      } else if (state.selectedPaymentMethod === PaymentTypes.BCOL) {
+        isValid = !!(state.bcolInfo.userId && state.bcolInfo.password)
+        if (!isValid) {
+          state.errorMessage = 'Missing User ID and Password for BC Online.'
+          state.isLoading = false
+        }
+        createRequestBody = {
+          paymentInfo: {
+            paymentMethod: PaymentTypes.BCOL
+          },
+          bcOnlineCredential: state.bcolInfo
+        }
+      } else {
+        isValid = true
+        createRequestBody = {
+          paymentInfo: {
+            paymentMethod: state.selectedPaymentMethod
+          }
         }
       }
-    } else if (this.selectedPaymentMethod === PaymentTypes.BCOL) {
-      isValid = !!(this.bcolInfo.userId && this.bcolInfo.password)
-      if (!isValid) {
-        this.errorMessage = 'Missing User ID and Password for BC Online.'
-        this.isLoading = false
-      }
-      createRequestBody = {
-        paymentInfo: {
-          paymentMethod: PaymentTypes.BCOL
-        },
-        bcOnlineCredential: this.bcolInfo
-      }
-    } else {
-      isValid = true
-      createRequestBody = {
-        paymentInfo: {
-          paymentMethod: this.selectedPaymentMethod
+
+      if (isValid) {
```

Comment:
Worth putting this in it's own function?

Reply:
added